lib/resin-lint.ts

```diff
@@ -131,25 +131,28 @@ const lintTsFiles = async function(
 
 	for (const file of files) {
 		let source = read(file);
+		linter.lint(
+			file,
+			source,
+			config as tslint.Configuration.IConfigurationFile,
+		);
 		if (prettier) {
 			if (autoFix) {
 				const newSource = prettier.format(source, prettierConfig);
 				if (source !== newSource) {
 					source = newSource;
 					fs.writeFileSync(file, source);
 				}
-			}
-			const isPrettified = prettier.check(source, prettierConfig);
-			if (!isPrettified) {
-				console.log(`Error: File ${file} hasn't been formatted with prettier`);
-				return 1;
+			} else {
```

Comment:
Well spotted: the 'else' clause probably reduces running time by half.

Reply:
It doesn't seem to have that large of an effect in reality, but it's still nice to save - I expect most of the time is in tslint and the typescript checking it needs to do